pkg/scale/encode.go

```diff
@@ -41,6 +41,7 @@ func Marshal(v interface{}) (b []byte, err error) {
 	}
 	err = es.marshal(v)
 	if err != nil {
+		err = fmt.Errorf("marshalling: %w", err)
 		return
```

Comment:
Since callers of `scale.Marshal` *should* wrap the error returned with `fmt.Errorf("scale marshaling: %w", err)`, I don't think there is a point wrapping `err` here with `fmt.Errorf("marshalling: %w", err)` since that would just add `marshalling` a second time.


```suggestion
		return err
```